userguide/payment/includes/payment-methods-flows.adoc

```diff
@@ -0,0 +1,93 @@
+=== Payment Method Flows Overview
+
+When an end-user enters his/her payment details (like card information) into a merchant website, these payment details need to be saved in Kill Bill in order for Kill Bill to be able to process recurring payments. Additionally, the payment details may need to be saved within the payment gateway so that the end user does not have to enter their payment details each time. 
+
+*Payment method flows* describe how a user's payment details gets added in Kill Bill and the payment gateway.
+
+The end-user's payment details are referred to as a *payment method* in Kill Bill. The https://github.com/killbill/killbill-api/blob/4ae1c343a593de937415e21feecb9f5405037fa3/src/main/java/org/killbill/billing/payment/api/PaymentApi.java#L541[PaymentApi#addPaymentMethod] is the entry point for saving a payment method in Kill Bill.
+
+Kill Bill supports three main payment method flows as follows:
+
+* *Client-side tokenization* - In this case, the end user's payment details are converted into a token and sent to Kill Bill.
+
+*  *Hosted Payment Page* - In this case, the merchant website redirects the end-user to a page on the payment gateway which collects the payment details. Kill Bill then invokes the payment gateway to obtain the payment method details.
+
+* *Server-side tokenization* - In this case, the end user's payment details are sent to Kill Bill which invokes the payment gateway to perform tokenization.
+
+The flow diagrams below explain these payment method flows. We consider the following actors:
+
+* *Browser*: The end-user sitting behind a browser and initiating the payment method flow
+
+* *Merchant Site*: customer facing web site which receives the order
+
+* *Kill Bill* - The Kill Bill system
+
+* *Checkout Servlet* - Servlet that initiates setting up the payment method. 
+
+* *Payment Plugin*: https://docs.killbill.io/latest/payment_plugin.html[Kill Bill plugin] corresponding to the payment gateway 
+
+* *Payment Gateway*: Entity that processes the payment
+
+=== Client-side Tokenization Payment Method Flow
+
+In this case, the merchant website collects the end-user's payment details. These are then converted into a *token* via a payment gateway specific Javascript which is embedded into the merchant site payment page. This token is sent to Kill Bill. Kill Bill then saves it in the Kill Bill database and uses it to charge the customer.
+
```

Comment:
Notes: In some cases instead of javascript integration, payment gateway providers provide  an iFrame instead. We could add this as a note if case folks have encountered it and get confused why this is not listed.